src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -400,9 +404,14 @@ class RedownloadService[F[_]: NonEmptyParallel](
           if (state.isNewSet) {
             wrappedRedownload(shouldPerformRedownload, meaningfulAcceptedSnapshots, meaningfulMajorityState).handleErrorWith {
               error =>
-                logger.error(error)(s"Redownload error: ${stringifyStackTrace(error)}") >> cluster
-                  .compareAndSet(NodeState.validDuringDownload, NodeState.Ready)
-                  .void
+                logger.error(error)(s"Redownload error, isDownload=$isDownload: ${stringifyStackTrace(error)}") >> {
+                  if (!isDownload)
+                    cluster
+                      .compareAndSet(NodeState.validDuringDownload, NodeState.Ready)
+                      .void
+                  else
+                    F.unit
+                } >> error.raiseError[F, Unit]
```

Comment:
should it throw also during redownload? is that handled in any other place?

Reply:
Initially I implemented it to only raise error for download case. Then I changed it to this one thinking that the next successful periodic run will cover what didn't finish here. But now looking at it I think again that it probably should let it finish gracefully when it's not during download/joining even if wrappedRedownload failed